tests/io/test_core.py

```diff
@@ -27,6 +32,13 @@
 ]
 
 
+@pytest.mark.parametrize("name", _DEPRECATED_ERROR_CLASSES)
+@pytest.mark.parametrize("package", ["kedro.io", "kedro.io.core"])
+def test_deprecation(name, package):
+    with pytest.warns(DeprecationWarning, match=f"{name} has been renamed"):
+        exec(f"from {package} import {name}")  # pylint: disable=exec-used
```

Comment:
Does this work if we use https://docs.python.org/3/library/importlib.html#importlib.import_module instead?

Reply:
No, it doesn't. :( `exec` was not even on my list of things to try (I didn't even know about it?), but nothing else seemed to work.

Reply:
Weird, we did some manual tests and seemed to work https://github.com/kedro-org/kedro/pull/2673#issuecomment-1598916317 in any case, if it doesn't, it's fine

Reply:
Oh, wow, I'm dumb. I tried `importlib.import_module("DataSetError", "kedro.io.core")` and stuff like that, but obviously you can't import a class using `import_module`. Fixing...